### ISSUE TITLE
Failure to parse request in Kitura-net

### ISSUE THREAD

**Issue by tfrank64** [IBM-Swift/Kitura#686]
In BluePic, we had this issue raised: https://github.com/IBM-Swift/BluePic/issues/306
I was able to reproduce the issue in BluePic and it consistently gives me the following error:
     
    ERROR: parse IncomingHTTPSocketProcessor.swift line 117 - Failed to parse a request

That error appears when trying to post an image in BluePic and sometimes when trying to login, preventing these actions from succeeding.

These errors only happen with the 0.24 releases of the Kitura suite, using the older 0.22 releases, everything works great. Since we've made no code changes, we are concluding right now that our packages are causing this error. For reference, here are the packages I tested that worked and didn't work:

**Succeeds**
```
    .Package(url: "https://github.com/IBM-Swift/Kitura.git", majorVersion: 0, minor: 22),
    .Package(url: "https://github.com/IBM-Swift/Kitura-CouchDB.git", majorVersion: 0, minor: 22),
    .Package(url: "https://github.com/IBM-Swift/Swift-cfenv.git", majorVersion: 1, minor: 4),
    .Package(url: "https://github.com/ibm-bluemix-mobile-services/bluemix-simple-http-client-swift.git", "0.4.1"),
    .Package(url: "https://github.com/ibm-bluemix-mobile-services/bluemix-objectstorage-swift-sdk.git", majorVersion: 0, minor: 5),
    .Package(url: "https://github.com/ibm-bluemix-mobile-services/bms-mca-kitura-credentials-plugin.git", "0.3.2"),
    .Package(url: "https://github.com/ibm-bluemix-mobile-services/bms-pushnotifications-serversdk-swift.git", majorVersion: 0, minor: 3)
```

**Fails**
```
    .Package(url: "https://github.com/IBM-Swift/Kitura.git", majorVersion: 0, minor: 24),
    .Package(url: "https://github.com/IBM-Swift/Kitura-CouchDB.git", majorVersion: 0, minor: 24),
    .Package(url: "https://github.com/IBM-Swift/Swift-cfenv.git", majorVersion: 1, minor: 4),
    .Package(url: "https://github.com/ibm-bluemix-mobile-services/bluemix-objectstorage-swift-sdk.git", majorVersion: 0, minor: 5),
    .Package(url: "https://github.com/ibm-bluemix-mobile-services/bms-mca-kitura-credentials-plugin.git", majorVersion: 0, minor: 3),
    .Package(url: "https://github.com/ibm-bluemix-mobile-services/bms-pushnotifications-serversdk-swift.git", majorVersion: 0, minor: 3)
```

**Comment by tfrank64** [comment]
I see, good to know. We will most likely revert to the 0.22 versions for now.

**Comment by shmuelk** [comment]
@tfrank64 Is it possible to get the URL that was failing to parse?

**Comment by shmuelk** [comment]
@tfrank64 Is it possible for me to easily set up locally enough to run the failing request?

**Comment by tfrank64** [comment]
At least one place it fails is with this endpoint `http://localhost:8090/users`. The app tries to create a user before posting an image.

**Comment by tfrank64** [comment]
@shmuelk there is a somewhat easy way to get setup, you can follow this wiki to get the right credentials in your local BluePic: https://github.com/IBM-Swift/BluePic/wiki/Configuration-Process-for-Internal-Developers. Also, make sure your Package.swift looks like the one that fails, mentioned above. After that, you should be able to run locally and see it fail when posting an image.

**Comment by tfrank64** [comment]
@shmuelk Any updates on this or were you able to reproduce the issue?

**Comment by shmuelk** [comment]
I had a lot of trouble compiling the iOS code. I have XCode 8 beta 6 installed. I donloaded XCode beta 3, but I haven't gotten around to installing it and trying to build your iOS code.

**Comment by ianpartridge** [comment]
@rob-deans Were you looking at this?

**Comment by tfrank64** [comment]
@ianpartridge I think the current plan is for us to update BluePic to the latest version of Kitura to test if the issue got resolved at some point.

**Comment by tfrank64** [comment]
@rob-deans So we updated BluePic-Server to use Kitura 0.32.0. Unfortunately, the issue persists when trying to post an image. The error is slightly different now:
`ERROR: parse IncomingHTTPSocketProcessor.swift line 124 - Failed to parse a request. Parsed fewer bytes than were passed to the HTTP parser `

Hitting the `http://localhost:8090/users` endpoint through something Postman works great.

**Comment by tfrank64** [comment]
To be clear, the issue happens when creating a user, which happens right before posting an image. For some reason I can get a curl request to work, but it fails when coming from the iOS app. Here is an example curl request:

    curl -X POST -H "Content-Type: application/json" -H "Cache-Control: no-cache" -d '{ "name": "<myName>", "_id": "<userID>" }' "http://localhost:8090/users"

**Comment by shmuelk** [comment]
I just tagged Kitura-net 1.0.1 which includes the fix for issue IBM-Swift/Kitura#734 which we believe fixes the issue here as well. Please test again, make sure that you refetch Packages.

**Comment by tfrank64** [comment]
@shmuelk @rob-deans I tested things out, seems to be working great now! Thanks
